### PR TITLE
Revert some changes for Windows build since they seem unnecessary

### PR DESCRIPTION
**Pull request by anmyachev**
Part of https://github.com/intel/intel-xpu-backend-for-triton/issues/2824

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -90,11 +90,7 @@ endif()
 # Compiler flags
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 if(NOT MSVC)
-  if(NOT WIN32)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STDC_FORMAT_MACROS -fPIC")
-  else()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STDC_FORMAT_MACROS -Wno-deprecated")
```
Comment (anmyachev) [inline CMakeLists.txt:96]: don't consider case: not MSVC + windows since it's not a target

File: CMakeLists.txt [hunk 2]
```diff
@@ -165,8 +161,7 @@ include_directories(${PROJECT_BINARY_DIR}/include) # Tablegen'd files
 include_directories(${PROJECT_SOURCE_DIR}/third_party)
 include_directories(${PROJECT_BINARY_DIR}/third_party) # Tablegen'd files
 
-link_directories(${LLVM_LIBRARY_DIR})
-
+# link_directories(${LLVM_LIBRARY_DIR})
```
Comment (anmyachev) [inline CMakeLists.txt:164]: Why do we need to link for Windows when we don't link for Unix platforms? (it's just a revert)

File: CMakeLists.txt (hunk 1 above)
Comment (pbchekin) [reply CMakeLists.txt:96]: May be this is to support building with llvm on windows?

**Review by pbchekin — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (anmyachev) [reply CMakeLists.txt:96]: I have a feeling that this is for `clang`. If anything, I'll revert it

**Pull request merged**